odoo/addons/base/models/res_partner.py

```diff
@@ -265,11 +265,11 @@ def _compute_avatar(self, avatar_field, image_field):
             partner[avatar_field] = partner[image_field] or partner._avatar_get_placeholder()
 
     def _avatar_get_placeholder_path(self):
-        if self.is_company:
+        if self and self.is_company:
```

Comment:
This change should not be necessary: if `self` is an empty recordset then `self.is_company` should be `False`, and so should `self.type` below.